earth_enterprise/src/fusion/autoingest/plugins.src/PacketGen.src

```diff
@@ -255,9 +255,16 @@ void PacketGenAssetVersionImplD::UpdateChildren(
           // assets) that have no alpha blend cached.
           levInputs[inputs_idx+1] = blendVer.Ref();
           levInputVers[inputs_idx+1] = blendVer;
+          const std::string output_file_name = blendVer->GetOutputFilename(0);
           const std::string cache_alpha_path =
-              blendVer->GetOutputFilename(0) + "/cache_alpha.pack";
-          if (khExists(cache_alpha_path)) {
+              output_file_name + "/cache_alpha.pack";
+          // if output_file_name is empty, this Asset still 
+          // doesn't has an output file (this is a new asset), so the path 
+          // doesn't exist. But since in our version alpha cache is going
+          // to be used (according to above comment), we assume that it's 
+          // going to be there when it would be needed.
+          if (output_file_name.empty() 
```

Comment:
I'm a bit concerned about this change. At this point, all the resources should be built, so if there is an alpha mask it should already exist. I don't know of a situation in which the alpha mask would appear after this point. I'm also worried that this will break some other situation, such as upgrades from 4.x. Can you provide a more detailed set of steps for reproducing this issue? I haven't been able to reproduce it yet, but if I could I might be able to suggest a safer fix.

Reply:
@tst-lsavoie thanks.
I hope I recall correctly, since the last time I touched this was 2 months ago.
I just want to be aligned on the terminology- this is an asset version of packetlevel asset (not resource), and if I recall correctly it's produced every time this resource or resources below it in the project are changed.
Unfortunately I didnt find the code for the 4.x version (the first version I managed to install from the git was 5.2 and it had this bug).
It's does seem that at this point the packetgen/packetlevel should already be built, but in fact the packetlevel (not only the alpha cache) doesn't exist at this stage on the disk and the asset doesn't has an output path. Maybe the task order for building process should be fixed.
Since this is a new asset version (otherwise the packetlevel  would have already been built and have an output path), which is now being built in version 5.x envireonment, it looked to me safe to continue even with 4.x projects, but this is the first time I'm getting into fusion source code so you probably will have a better fix.

I wrote more details about how to reproduce this issue on the issue itself.